src/main/java/bio/terra/service/resourcemanagement/azure/AzureMonitoringService.java

```diff
@@ -0,0 +1,626 @@
+  private static final String UNAUTHORIZED_ACCESS_ALERT_NAME = "UnauthorizedAccess";
+  private static final String SENTINEL_ONBOARD_STATE_NAME = "default";
+  private static final String SLACK_ALERT_RULE_NAME = "runSendSlackNotificationPlaybook";
+  // Leaving this as static as opposed to config since these do not change from one environment to
+  // the next
+  private static final List<String> TABLES_TO_EXPORT =
+      List.of(
+          "Alert",
+          "AppCenterError",
+          "AzureMetrics",
+          "ComputerGroup",
+          "InsightsMetrics",
+          "Operation",
+          "StorageBlobLogs",
+          "Usage");
+
+  private final AzureResourceConfiguration resourceConfiguration;
+
+  @Autowired
+  public AzureMonitoringService(AzureResourceConfiguration resourceConfiguration) {
+    this.resourceConfiguration = resourceConfiguration;
+  }
+
+  /**
+   * Retrieve an existing Log Analytics workspace
+   *
+   * @param profileModel The billing profile for the dataset or snapshot being logged
+   * @param storageAccount The Storage Account being logged
+   * @return A {@link Workspace} object representing the Log Analytics workspace, or null if it does
+   *     not exist
+   */
+  public Workspace getLogAnalyticsWorkspace(
+      BillingProfileModel profileModel, AzureStorageAccountResource storageAccount) {
+    LogAnalyticsManager client =
+        resourceConfiguration.getLogAnalyticsManagerClient(
+            resourceConfiguration.getCredentials().getHomeTenantId(),
+            profileModel.getSubscriptionId());
+    try {
+      Workspace byResourceGroup =
+          client
+              .workspaces()
+              .getByResourceGroup(
+                  storageAccount.getApplicationResource().getAzureResourceGroupName(),
+                  storageAccount.getName());
+      logger.info("Found Log Analytics Workspace");
+      return byResourceGroup;
+    } catch (ManagementException e) {
+      logger.warn("No Log Analytics Workspace found", e);
+      if (Objects.equals(e.getValue().getCode(), RESOURCE_NOT_FOUND_CODE)) {
+        return null;
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  /**
+   * Create a new Log Analytics workspace
+   *
+   * @param profileModel The billing profile for the dataset or snapshot being logged
+   * @param storageAccount The Storage Account being logged
+   * @return A string representing the id of the Log Analytics workspace that was created
+   */
+  public String createLogAnalyticsWorkspace(
+      BillingProfileModel profileModel, AzureStorageAccountResource storageAccount) {
+    LogAnalyticsManager client =
+        resourceConfiguration.getLogAnalyticsManagerClient(
+            resourceConfiguration.getCredentials().getHomeTenantId(),
+            profileModel.getSubscriptionId());
+
+    logger.info(
+        "Creating new Log Analytics Workspace for Storage Account {}",
+        storageAccount.getStorageAccountId());
```

Comment:
Why do you log the "id", which is the path `/subscriptions/<subscriptionId>/resourceGroups/<resourceGroup>/providers/Microsoft.Storage/storageAccounts/<storageAccount>` instead of logging `storageAccount.getName()`?

Reply:
I've been spending a lot of time recreating IDs so I like the idea of getting a string (where we can) that includes more information.

Reply:
OK. TDR will probably never have a deployment that disables INFO so I guess there's no need to worry, but in general if you call a method in the parameter list to a log API, and that method isn't a simple getter (this method calls UUID.fromString(), ST.render(), etc) then it's good practice to wrap the log call in a check to see if the logger is enabled to avoid unnecessary overhead when it's not enabled:
```
    if (logger.isInfoEnabled()) {
      logger.info(
          "Creating new Log Analytics Workspace for Storage Account {}",
          storageAccount.getStorageAccountId());
    }
```